ompi/mpi/fortran/base/strings.c

```diff
@@ -95,8 +96,19 @@ int ompi_fortran_string_c2f(const char *cstr, char *fstr, int len)
     int i;
 
     opal_string_copy(fstr, cstr, len);
-    for (i = strlen(cstr); i < len; ++i) {
-        fstr[i] = ' ';
+
+    // If len < len(cstr), then opal_string_copy() will have copied a
+    // trailing \0 into the last position in fstr.  This is not what
+    // Fortran wants; overwrite that \0 with the actual last character
+    // that will fit into fstr.
+    if (len < strlen(cstr)) {
```

Comment:
Thanks for looking at this Jeff! Should this be `<=` though? I think the case where they're equal still results in the last character being overwritten with a null.